lib/plugins/aws/deploy/lib/check-for-changes.js

```diff
@@ -25,25 +24,19 @@ module.exports = {
       return this.checkLogGroupSubscriptionFilterResourceLimitExceeded();
     }
 
-    return BbPromise.bind(this)
-      .then(this.getMostRecentObjects)
-      .then((objs) => {
-        return BbPromise.all([
-          this.getObjectMetadata(objs),
-          this.getFunctionsEarliestLastModifiedDate(),
-        ]);
-      })
-      .then(([objMetadata, lastModifiedDate]) =>
-        this.checkIfDeploymentIsNecessary(objMetadata, lastModifiedDate)
-      )
-      .then(() => {
-        if (this.serverless.service.provider.shouldNotDeploy) {
-          return BbPromise.resolve();
-        }
+    const objs = await this.getMostRecentObjects();
+    const [objMetadata, lastModifiedDate] = await Promise.all([
+      this.getObjectMetadata(objs),
+      this.getFunctionsEarliestLastModifiedDate(),
+    ]);
+    await this.checkIfDeploymentIsNecessary(objMetadata, lastModifiedDate);
+
+    if (this.serverless.service.provider.shouldNotDeploy) {
+      return Promise.resolve();
```

Comment:
Let's simply `return` (no need for `Promise.resolve()`

Reply:
I am getting an eslint error if I were to return <empty>
https://eslint.org/docs/latest/rules/consistent-return

Reply:
Yes, the function either needs to return value or not.

As this function is not designed to return specific value, I think fix is to ensure we do not return last promise but just await it

Reply:
it does return another promise, let me check further if we can turn that into an `await` statement instead. In that case we can remove the `resolve()` call